airbnb_project/listings/harvester_app/harvester/spiders/listings_spider.py

```diff
@@ -0,0 +1,312 @@
+                    row: dict
+                    ListingsSpider.ALREADY_SCRAPED_LISTINGS.add(row['airbnb_listing_id'])
+        except FileNotFoundError as e:
+            print(e)
+        requests = []
+        try:
+            with open('cordinates.json', 'r') as file:
+                coordinates = json.load(file)
+                urls = ListingsSpider.URL_BUILDER.get_urls("july", "june", "august")
+                if coordinates is not None:
+                    requests = [scrapy.FormRequest(
+                        url.format(cord["ne_lat"], cord["ne_lng"], cord["sw_lat"], cord["sw_lng"],
+                                   ZOOM_LEVEL,
+                                   ZOOM_LEVEL)) for url in urls for
+                        cord in
+                        coordinates]
+        except FileNotFoundError:
+            print("The file was not found")
+        except json.JSONDecodeError:
+            print("Error decoding JSON")
+        return requests
+
+    async def parse(self, response: Response, **kwargs):
+        """
+        Parse method for extracting listings from the Airbnb search results page.
+
+        Args:
+            response (Response): The response object from the request.
+
+        Yields:
+            Request: A request object for each listing's details page.
+        """
+        if response.status != 200:
+            return
+        script_tag = response.css(f'script#{SCRIPT_TAG}')
+        script_inner_text = script_tag.css('script::text').get()
+        script_json = json.loads(script_inner_text)
+        results = self._parse_listings_json(script_json)
+        current_url = response.url
+
+        if ListingsSpider.next_page_cursors is None:
+            ListingsSpider.next_page_cursors = self._get_cursors(script_json)
+
+        for result in results:
+            try:
+                listing = result.get("listing", {})
+                listing_id: str = listing.get("id")
+                print("listing_id", listing_id)
+                title = listing.get("title", "")
+                name = listing.get("name", "")
+                coordinate = listing.get("coordinate", {})
+                latitude = coordinate.get("latitude", "")
+                longitude = coordinate.get("longitude", "")
+                room_type = listing.get("roomTypeCategory", "")
+                page_url = API_CALL.format(n(t("StayListing", listing_id)))
+                airbnb_params = {
+                    "airbnb_listing_id": listing_id,
+                    "title": title,
+                    "name": name,
+                    "latitude": latitude,
+                    "longitude": longitude,
+                    "room_type": room_type
+                }
+
+                # A second request is made to get the registration number and other information using the listing_id
+                # which is scarped
+                if listing_id and listing_id not in ListingsSpider.ALREADY_SCRAPED_LISTINGS:
```

Comment:
duplicated `listing_id`

Reply:
`listing_id`  is supposed to be duplicated. It check if its `truthy` and if yes, then checks if the id has already been scraped.

Reply:
i saw that listing_id was used in print statement line 142 and in `airbnb_params`, so does truthiness needs to be checked before all this?